miden-lib/asm/kernels/transaction/lib/tx.masm

```diff
@@ -178,6 +184,54 @@ proc.add_non_fungible_asset_to_note
     # => [note_ptr, note_idx]
 end
 
+#! Updates the transaction expiration block number.
+#!
+#! The input block_height_delta is added to the block reference number in order to output an upper
+#! limit at which the transaction will be considered valid (not expired).
+#! This value can be later decreased, but not increased. 
+#!
+#! Inputs: [block_height_delta, ...]
+#! Output: [...]
+#!
+#! Where:
+#! - block_height_delta is the desired expiration time delta (1 to 0xFFFF).
+#! - tx_block_height_delta is the updated (or unchanged) transaction expiration time delta.
+export.update_expiration_block_num
+    # Ensure block_height_delta is between 1 and 0xFFFF (inclusive)
+    dup neq.0 assert.err=ERR_INVALID_TX_EXPIRATION_DELTA
+    dup push.EXPIRY_UPPER_LIMIT lt assert.err=ERR_INVALID_TX_EXPIRATION_DELTA
+    # => [block_height_delta]
+
+    exec.get_block_number add
+    # => [absolute_expiration_num]
+
+    # Load the current stored delta from memory
+    dup exec.memory::get_expiration_block_num  
+    # => [block_height_delta, absolute_expiration_num, absolute_expiration_num]
+
+    # Check if block_height_delta is greater
+    lt
+    if.true
+        # Set new expiration delta
+        exec.memory::set_expiration_block_num 
+    else
+        drop                           
+    end
+end
+
+#! Gets the transaction expiration block number.
+#!
+#! Inputs: [...]
+#! Output: [block_height_delta, ...]
+#!
+#! Where:
+#! - block_height_delta is the stored expiration time delta (1 to 0xFFFF).
+export.get_expiration_delta
```

Comment:
Line 222 needs to be updated since we are returning the delta - not the expiration block number.